### ISSUE TITLE
[MAINT] [ENH] HPC feedback/improvements

### ISSUE THREAD

**Issue by michellewang** [nipoppy/nipoppy#455]
Potential TODOs after #437 is merged (list to be updated as this is tested by more people):

- [ ] **[IMPORTANT]** Documentation: how-to guide for running pipelines with Nipoppy on a high-performance computing system
- [x] Include paths to HPC templates and logs in `DatasetLayout` class (but make logs directory optional)
- [x] Some validation in `HpcConfig` to make sure that there are no collisions with e.g. `dataset_root` or `command`
  - Maybe prepending the args passed by Nipoppy with `nipoppy_` and leave others as-is would help
- [x] Remove warning "no participants/sessions to run"
- [x] **[IMPORTANT]** Propagation of optional CLI arguments (e.g. `--tar`, `--keep-workdir`) #497
- [x] (Maybe) change `HPC_PREAMBLE` to be list of strings
- [x] (Maybe) add optional configs to templates based on Slurm setup of collaborators (e.g. `partition`)
- [x] (Maybe) modify template files a bit
  - [x] Is it possible to have only 1 template file?
  - [x] More flexibility in specifying time/memory requirements
- [x] (Maybe) change default configs -- seems like a lot of memory is used for fMRIPrep, and I'm unsure how many cores are needed by MRIQC
- [x] (Maybe, if possible) make PySQA write `run_queue.sh` in `<DATASET_ROOT>/logs/hpc`
  - Looks like it might be possible to set to the working directory, which could be helpful. Unsure if we want this file in work dir or somewhere else

**Comment by mathdugre** [comment]
- handling empty `HPC_PREAMBLE`.
- Default pysqa in queue are overwriting global_config values.
- empty slurm_account should be handled

Nice to have
- Look if we can merge the slurm and sge template files
- Split bash -c command over multiple line (at least one per command)
- Prevent queueing if job already exist -> Check if job name already exist (must be unique for each subject-pipeline pair)

**Comment by michellewang** [comment]
- [ ] Once the global config changes are merged to main, we can move the `HPC_CONFIG` part out of the pipeline config and into an `hpc_config.json` file
